__tests__/index.spec.js

```diff
@@ -1,18 +1,253 @@
+        expect(reporter.reportOptions).toBeDefined();
+        expect(reporter.client).toBeDefined();
+    });
+
+    test('onRunStart: startLaunch should be called with parameters', () => {
+        reporter.onRunStart();
+
+        expect(reporter.tempLaunchId).toEqual('startLaunch');
+        expect(reporter.client.startLaunch).toHaveBeenCalledTimes(1);
+
+        const launchObj = {
+            launch: options.launch,
+            description: options.description,
+            attributes: options.attributes,
+            rerun: undefined,
+            rerunOf: undefined,
+            startTime: new Date().valueOf()
+        };
+
+        expect(reporter.client.startLaunch).toHaveBeenCalledWith(launchObj);
+    });
+
+    test('onTestResult: startSuite, startTest, finishTest, finishSuite should be called with parameters', () => {
+        const spyStartSuite = jest.spyOn(reporter, '_startSuite');
+        const spyStartTest = jest.spyOn(reporter, '_startTest');
+        const spyFinishTest = jest.spyOn(reporter, '_finishTest');
+        const spyFinishSuite = jest.spyOn(reporter, '_finishSuite');
+
+        reporter.onTestResult({}, testResult);
+
+        expect(spyStartSuite).toHaveBeenCalledWith(testResult.testResults[0].ancestorTitles[0]);
+        expect(spyStartTest).toHaveBeenCalledWith(testResult.testResults[0].title, false);
+        expect(spyFinishTest).toHaveBeenCalledWith(testResult.testResults[0], false);
+        expect(spyFinishSuite).toHaveBeenCalled();
+    });
+
+    test('onTestResult: startTest, finishTest should be called one times with second parameter \'false\' if there are no retries', () => {
+        const spyStartTest = jest.spyOn(reporter, '_startTest');
+        const spyFinishTest = jest.spyOn(reporter, '_finishTest');
+
+        reporter.onTestResult({}, testResult);
+
+        expect(spyStartTest).toHaveBeenCalledWith(testResult.testResults[0].title, false);
+        expect(spyFinishTest).toHaveBeenCalledWith(testResult.testResults[0], false);
+        expect(spyStartTest).toHaveBeenCalledTimes(1);
+        expect(spyFinishTest).toHaveBeenCalledTimes(1);
+    });
+
+    test('onTestResult: startTest, finishTest should be called two times with second parameter \'true\' if there are retries', () => {
+        const spyStartTest = jest.spyOn(reporter, '_startTest');
+        const spyFinishTest = jest.spyOn(reporter, '_finishTest');
+        testResult.testResults[0].invocations = 2;
+
+        reporter.onTestResult({}, testResult);
+
+        expect(spyStartTest).toHaveBeenCalledWith(testResult.testResults[0].title, true);
+        expect(spyFinishTest).toHaveBeenCalledWith(testResult.testResults[0], true);
+        expect(spyStartTest).toHaveBeenCalledTimes(2);
+        expect(spyFinishTest).toHaveBeenCalledTimes(2);
+    });
+
+    test('onRunComplete: finishLaunch should be called with tempLaunchId', () => {
+        reporter.tempLaunchId = 'tempLaunchId';
```

Comment:
The same for changing global state

Reply:
This is a necessary variable assignment, I do not see another way

Reply:
Just reset such values after tests in corresponding "after" methods.